src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs

```diff
@@ -273,6 +274,33 @@ public TimeSpan Expect100ContinueTimeout
             }
         }
 
+        /// <summary>
+        /// When non-null, a custom factory used to open new TCP connections.
+        /// When null, a <see cref="SocketsHttpConnectionFactory"/> will be used.
+        /// </summary>
+        public ConnectionFactory? ConnectionFactory
```

Comment:
Why is this nullable? Why don't we just return the default factory instead of null?

Reply:
IIRC no strong reasons, it's just the way we decided to go.